debug/debug.proto

```diff
@@ -0,0 +1,77 @@
+//
+// Copyright 2023 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// This file defines the gNOI APIs used to gather specific debug information
+// from the network device.
+syntax = "proto3";
+
+package gnoi.debug;
+
+import "google/protobuf/any.proto";
+import "types/types.proto";
+
+option go_package = "github.com/openconfig/gnoi/debug";
+
+option (types.gnoi_version) = "0.1.0";
+
+service Debug {
+  // Debug will execute the set of commands provided in the request.
+  // The command will be executed in the mode provided.
+  // All command modes must support an exit code on completion of the
+  // command. (e.g. Cli modes must exit after sending a command)
+  // Errors:
+  //  InvalidArgument: for unspecified mode
+  rpc Debug (DebugRequest) returns (stream DebugResponse);
+}
+
+message DebugRequest {
+  enum Mode {
+    MODE_UNSPECIFIED = 0;
+    MODE_SHELL = 1;
+    MODE_CLI = 2;
+  }
+  // Mode the commands will be executed in.
+  Mode mode = 1;
+  // Raw bytes for the command to be executed.
+  bytes command = 2;
```

Comment:
Why did we make this `bytes`? Is there a case where it's ever something other than a string? I'm wondering whether having `bytes` here introduces any new attack vector vs. having a string that could be parsed out (I presume I can't just put an statically linked executable in this field ;-))

Reply:
i could do string i think it won't matter

Reply:
After our discussion, I'm OK with `bytes`.